gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java

```diff
@@ -31,30 +32,63 @@
 import org.gradle.api.JavaVersion;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.file.FileCollection;
+import org.gradle.api.file.RegularFile;
 import org.gradle.api.plugins.ExtensionAware;
+import org.gradle.api.provider.Provider;
 import org.gradle.api.tasks.compile.JavaCompile;
 import org.gradle.api.tasks.javadoc.Javadoc;
 import org.gradle.api.tasks.testing.Test;
 
 public final class BaselineErrorProne implements Plugin<Project> {
 
+    public static final String REFASTER_CONFIGURATION = "refaster";
     public static final String EXTENSION_NAME = "baselineErrorProne";
 
     private static final String ERROR_PRONE_JAVAC_VERSION = "9+181-r4173-1";
     private static final String PROP_ERROR_PRONE_APPLY = "errorProneApply";
+    private static final String PROP_REFASTER_APPLY = "refasterApply";
 
     @Override
     public void apply(Project project) {
         project.getPluginManager().withPlugin("java", plugin -> {
             BaselineErrorProneExtension errorProneExtension = project.getExtensions()
                     .create(EXTENSION_NAME, BaselineErrorProneExtension.class, project);
             project.getPluginManager().apply(ErrorPronePlugin.class);
+
             String version = Optional.ofNullable(getClass().getPackage().getImplementationVersion())
                     .orElse("latest.release");
+
+            Configuration refasterConfiguration = project.getConfigurations().create(REFASTER_CONFIGURATION);
+            Configuration refasterSources = project.getConfigurations()
+                    .create("refasterSources", configuration -> {
+                        configuration.extendsFrom(refasterConfiguration);
+                        configuration.setTransitive(false);
+                    });
+            Configuration refasterCompilerConfiguration = project.getConfigurations()
+                    .create("refasterCompiler", configuration -> configuration.extendsFrom(refasterConfiguration));
+
+            project.getDependencies().add(
+                    REFASTER_CONFIGURATION,
+                    "com.palantir.baseline:baseline-refaster-rules:" + version + ":sources");
```

Comment:
I think it might not actually be necessary to ship this to everyone as the _vast_ majority of builds will not use `-PrefasterApply`

Reply:
If they don't use refasterApply, then no harm in adding it to the refaster configuration right?

Reply:
Fair enough - we've stopped using `resolveConfigurations` in our CircleCI templates, so should be fine!